receiver.go

```diff
@@ -38,50 +38,54 @@ func (r *Receiver) DrainCredit(ctx context.Context) error {
 	return r.link.DrainCredit(ctx)
 }
 
-// Receive returns the next message from the sender.
+// ReceiveCached returns the next message that is stored in the Receiver's
+// internal message channel. It does NOT wait for the remote sender to send messages
+// and returns immediately if the internal cache is empty.
 //
-// Blocks until a message is received, ctx completes, or an error occurs.
-// When using ModeSecond, you *must* take an action on the message by calling
-// one of the following: AcceptMessage, RejectMessage, ReleaseMessage, ModifyMessage.
-// When using ModeFirst, the message is spontaneously Accepted at reception.
-func (r *Receiver) Receive(ctx context.Context) (*Message, error) {
+// NOTE: Most callers will use `Receive`, which checks both the internal channel and also
+// waits for messages to arrive from the remote Sender.
+func (r *Receiver) ReceiveCached(ctx context.Context) (*Message, error) {
```

Comment:
Feels like the name could be improved as we're not receiving anything.  GetCached?

Reply:
Updated docs, went with GetCached

Reply:
Updated again, went with `Prefetched`